prototypes/ScheduledMerges.hs

```diff
@@ -367,11 +375,12 @@ invariant (LSMContent _ levels ul) = do
         MergePolicyLevelling -> do
           case (ir, mrs) of
             -- A single incoming run (which thus didn't need merging) must be
-            -- of the expected size range already
+            -- of the expected size range already, but it could also be smaller
+            -- if it comes from a union level.
             (Single r, m) -> do
               assertST $ case m of CompletedMerge{} -> True
                                    OngoingMerge{}   -> False
-              assertST $ levellingRunSizeToLevel r == ln
+              assertST $ levellingRunSizeToLevel r <= ln
```

Comment:
We can make the check a little bit stricter if we distinguish between a `Single` run and a `MigratedUnion` run. In the former case we can then check equality, and for the latter case check inequality.

It might be nice to make that distinction anyway. It does not come at much of a cost, I think?

Reply:
It requires an extra case in a few pattern matches, but that's not really an issue. Stricter invariants are probably worth it.